shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -418,6 +418,7 @@ private String maybeGetInitialRouteFromIntent(Intent intent) {
   void onResume() {
     Log.v(TAG, "onResume()");
     ensureAlive();
+    flutterEngine.enableGpu();
```

Comment:
This adds another manual step users that don't use the activity or fragment have to remember to do (such as FlutterView integrators). Since it seems like the implementation it's meant to be platform agnostic, can't this hook into the animator.cc to keep everything in the same place instead of double book keeping?

Reply:
The problem is that this is a separate concern the animator doesn't know anything about. For example, we might be using a software backend and never use the GPU either way. Or, we might be on a platform that doesn't care if you use the GPU in the background.

Reply:
Actually you're right. I think your approach is mostly consistent with https://github.com/flutter/engine/pull/13908/files#diff-d98d87c5024ca62c96bfb181b13b8bb41f8f4c8bb218f102f7ba3bc8fc601af5R655 (considering that there isn't a way to get application-wide foreground-ness). 

It's also generally consistent with our previous discussion at https://github.com/flutter/flutter/issues/45131#issuecomment-555750287. SG then.

Reply:
Another thing you can try is since on Android, since it's so much more tied to the surface creation/destruction (which happens on the first onDraw of the view after attaching to a window) than the activity's resume/pause which is slightly orthogonal, we might be able to tie it to just the existing flutterJNI's onSurfaceDestroyed. It's already tied to the SurfaceHolder.Callback's surfaceDestroy which happens normally with activity onPause and is more semantically tied with what you're trying to do. And you get the bonus of having it work consistently with activity/fragment/view without adding new API surface area.

Reply:
I'd be curious if that causes any different problems - for example, the motivation of this patch is to fix images when they're initially decoded in the background, because they no longer render at all once you come to the foreground if they were directly uploaded to the GPU in the background. I'm not sure the same is true if you're in the foreground but don't yet have a surface ready. If it is this would be a good decision, but if it's not this would potentially increase jank unnecessarily.